otlp/otlp_grpc_metrics_client.cc

```diff
@@ -38,4 +43,67 @@ grpc::Status OtlpGrpcMetricsClient::Export(ExportMetricsServiceRequest const &re
   return stub_->Export(&context, request, &response);
 }
 
-} /* namespace otlp */
+void OtlpGrpcMetricsClient::AsyncExport(ExportMetricsServiceRequest const &request)
+{
+  SCOPED_TIMING(OtlpGrpcMetricsClientAsyncExport);
+
+  u64 async_response_tag = next_async_response_tag_++;
+
+  auto async_response = std::make_unique<AsyncResponse>();
+  async_response->response_reader_ = stub_->PrepareAsyncExport(&async_response->context_, request, &cq_);
+  async_response->response_reader_->StartCall();
+
+  async_response->response_reader_->Finish(
+      &async_response->response_, &async_response->status_, reinterpret_cast<void *>(async_response_tag));
+
+  async_responses_.emplace(async_response_tag, std::move(async_response));
+
+  ++requests_sent_;
+  bytes_sent_ += request.ByteSizeLong();
+  metrics_sent_ += request.resource_metrics(0).scope_metrics(0).metrics_size();
+}
+
+void OtlpGrpcMetricsClient::process_async_responses()
+{
+  if (async_responses_.empty()) {
+    return;
+  }
+
+  void *tag;
+  bool ok;
+  while (true) {
+    switch (cq_.AsyncNext(&tag, &ok, gpr_time_0(GPR_CLOCK_MONOTONIC))) {
+    case grpc::CompletionQueue::TIMEOUT:
+    case grpc::CompletionQueue::SHUTDOWN:
+      // no responses to process
+      return;
+      break;
+    case grpc::CompletionQueue::GOT_EVENT:
+      // Note: Per gRPC docs, for client-side Finish: ok should always be true, so not checking it here.
+
+      u64 u64_tag = reinterpret_cast<u64>(tag);
+
+      auto itr = async_responses_.find(u64_tag);
```

Comment:
I'd move the `itr` declaration and initialization inside the if statement: `if (itr = async...; itr == ...`